datafusion/sqllogictest/test_files/sort_merge_join.slt

```diff
@@ -508,6 +508,124 @@ select t1.* from t1 where not exists (select 1 from t2 where t2.a = t1.a and t1.
 ----
 
 
+# RIGHTSEMI join tests
```

Comment:
Im not sure these tests makes a difference as RIGHT SEMI are not supported by SQL this type of join can be called by optimizer on physical executiion